scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/cipher/CipherSuite.java

```diff
@@ -743,12 +745,38 @@ public boolean isSupported() {
 			return supported;
 		}
 
+		/**
+		 * Gets the thread local MAC used by this MAC algorithm.
+		 * 
+		 * Calls {@link Mac#reset()} on access.
+		 * 
+		 * @return mac, or {@code null}, if not supported by vm.
+		 */
 		public Mac getMac() {
-			return mac == null ? null : mac.current();
+			if (mac != null) {
+				Mac current = mac.current();
+				current.reset();
```

Comment:
was it a bug ? (the missing reset ?)

Reply:
I added it for cases, where something fails before and so the `doFinal` or `digest` is not called. But I didn't recognized, that actually this happens.